src/runtime/pkg/oci/utils.go

```diff
@@ -611,6 +611,16 @@ func addHypervisorMemoryOverrides(ocispec specs.Spec, sbConfig *vc.SandboxConfig
 		return err
 	}
 
+	// HugepageSize valid only HugePages is true
+	if value, ok := ocispec.Annotations[vcAnnotations.HugepageSize]; sbConfig.HypervisorConfig.HugePages && ok {
+		hugepageTypes := vc.GetHugepageTypes()
+		for _, t := range hugepageTypes {
+			if value == t {
+				sbConfig.HypervisorConfig.HugepageSize = value
+			}
+		}
```

Comment:
With the `isHugepageSizeValid()` function proposed below, this would reduce to

```
if isHugepageSizeValid(value) {
	sbConfig.HypervisorConfig.HugepageSize = value
}
```
right?